tests/test_feature_append_rename.py

```diff
@@ -129,9 +129,20 @@ def check_meta_is(iamdf, meta_col, val):
     check_meta_is(res, "col_2", "bye")
 
 
-def test_append_duplicates(test_df_year):
+def test_append_duplicates_raises(test_df_year):
+    # Merging objects with overlapping values (merge conflict) raises an error
     other = copy.deepcopy(test_df_year)
-    pytest.raises(ValueError, test_df_year.append, other=other)
+    with pytest.raises(ValueError, match='Indexes have overlapping values:'):
+        test_df_year.append(other=other)
+
+
+def test_append_incompatible_col_raises(test_pd_df):
+    # Merging objects with different data index dimensions raises an error
+    df = IamDataFrame(test_pd_df)
+    test_pd_df['foo'] = 'baz'
+    other = IamDataFrame(test_pd_df)
+    with pytest.raises(ValueError, match='Incompatible timeseries data index'):
+        df.append(other=other)
```

Comment:
This doesn't actually test the original problem, which only occurred when inplace=True was added. I would test for both with and without explicitly.